_episodes/l1-05-test_doubles.md

```diff
@@ -0,0 +1,135 @@
+---
+title: "Tests doubles and dependency injection"
+teaching: TBC
+exercises: TBC
+questions:
+- "TBC"
+objectives:
+- "TBC"
+keypoints:
+- "TBC"
+---
+
+## Testing untestable code
+
+Sooner or later you will face a piece of code that is not straight forward to write a test for. It might be because it calls a function that requests some data from a piece of hardware. Or because it needs to access a database that it is not available in the testing environment. Or simply because it triggers a complex and time consuming computation process that is only suited to run in a supercomputer. Whatever the reason, you will be in trouble. Moreover, you might want to test if some intermediate result of the calculation, and not just the final output, is a valid one.
```

Comment:
```suggestion
Sooner or later you will face a piece of code that is not straightforward to write a test for. It might be because it calls a function that requests some data from a piece of hardware, or because it needs to access a database that it is not available in the testing environment, or simply because it triggers a complex and time consuming computation process that is only suited to run in a supercomputer. Whatever the reason, you have a problem. Moreover, you might want to test if some intermediate result in the calculation is valid, and not just the final output.
```